modules/managers/galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos3270.manager/src/main/java/dev/galasa/zos3270/spi/Terminal.java

```diff
@@ -155,16 +181,36 @@ public void disconnect() throws TerminalInterruptedException {
             network.close();
             logger.trace("Network closed OK");
         }
-        if (networkThread != null) {
+
+        logger.trace("connect: Thread "+Thread.currentThread().hashCode()+" is about to request the lock for the network thread");
+        synchronized(networkThreadLock) {
+            logger.trace("connect: Thread "+Thread.currentThread().hashCode()+" has acquired the lock for the network thread");
             try {
-                logger.trace("Waiting for network thread to end...");
-                networkThread.join();
-                logger.trace("Network thread ended OK");
-            } catch (InterruptedException e) {
-                logger.warn("Interrupted while waiting for network thread to end. Network thread might still be active.");
-                throw new TerminalInterruptedException("Join of the network thread was interrupted",e);
+                networkThread = new NetworkThread(this, screen, network, network.getInputStream(), this.deviceTypes);
+                networkThread.start();
```

Comment:
Curious about this (unsure if it's intentional) - do we need to start a new networkThread when disconnecting?

Reply:
It looked like there was re-connect logic in there also. So I thought a belt-and-braces approach to protect all uses of the variable consistently would be safer.